src/Apollo.cpp

```diff
@@ -544,102 +518,115 @@ Apollo::train(int step) {
                     fout.close();
             }
 
-            // TODO(cdw): Load prior decisiontree...
-            reg->model = ModelFactory::createDecisionTree(
-                    num_policies,
-                    train_features,
-                    train_responses );
+            reg->model =
+                ModelFactory::createTuningModel(reg->model_name,
+                                                 num_policies,
+                                                 train_features,
+                                                 train_responses,
+                                                 reg->model_params);
+
 
             if (Config::APOLLO_RETRAIN_ENABLE)
+#ifdef ENABLE_OPENCV
               reg->time_model =
                   ModelFactory::createRegressionTree(train_time_features,
                                                      train_time_responses);
+#else
+              assert(false && "Retraining requires OpenCV");
```

Comment:
This assert could be compiled out, so mabye throw an exception instead?

Reply:
I agree that handling is not ideal. The ifdef-conditional assert is intentional. We do not (yet) support regression in our built-in ML library. The assertion will abort execution if the user enables re-training through the env var but has not compiled with ENABLE_OPENCV. Could we accept as a stopgap until regression falls in?

Reply:
Fixed through exceptions